source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -4,67 +4,111 @@
+            <VulID id="V-79119" Status="NotAFinding">
+                <Details>See System Security Document.</Details>
+                <Comments>This is a test of a manual check file entry.</Comments>
+            </VulID>
+        </ManualChecklistEntries>
+        
+        See a sample at /PowerShell/StigData/Samples/ManualCheckListEntriesExcelExport.xml.
 
     .EXAMPLE
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -ManualCheckFile "C:\Stig\ManualChecks\2012R2-MS-1.7.psd1" -XccdfPath $xccdfPath -OutputPath $outputPath
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -ManualCheckFile $manualCheckFilePath -XccdfPath $xccdfPath -OutputPath $outputPath
+        New-StigCheckList -MofFile $mofFile -XccdfPath $xccdfPath -OutputPath $outputPath -ManualChecklistEntries $manualChecklistEntriesFile
+        New-StigCheckList -MofFile $mofFile -ChecklistSTIGFiles $checklistSTIGFiles -OutputPath $outputPath -ManualChecklistEntries $manualChecklistEntriesFile
+        New-StigCheckList -DscResults $auditRehydrated -XccdfPath $xccdfPath -OutputPath $outputPath -ManualChecklistEntries $manualChecklistEntriesFile
+        New-StigCheckList -DscResults $auditRehydrated -ChecklistSTIGFiles $checklistSTIGFiles -OutputPath $outputPath -ManualChecklistEntries $manualChecklistEntriesFile
 #>
 function New-StigCheckList
 {
     [CmdletBinding()]
     [OutputType([xml])]
     param
     (
-        [Parameter(Mandatory = $true, ParameterSetName = 'mof')]
+        [Parameter(Mandatory = $true, ParameterSetName = 'single-mof')]
+        [Parameter(Mandatory = $true, ParameterSetName = 'multi-mof')]
         [string]
-        $ReferenceConfiguration,
+        $mofFile,
 
-        [Parameter(Mandatory = $true, ParameterSetName = 'result')]
+        [Parameter(Mandatory = $true, ParameterSetName = 'single-dsc')]
+        [Parameter(Mandatory = $true, ParameterSetName = 'multi-dsc')]
         [psobject]
-        $DscResult,
+        $DscResults,
 
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true, ParameterSetName = 'single-mof')]
+        [Parameter(Mandatory = $true, ParameterSetName = 'single-dsc')]
         [string]
-        $XccdfPath,
+        $xccdfPath,
 
-        [Parameter(Mandatory = $true)]
-        [System.IO.FileInfo]
-        $OutputPath,
+        [Parameter(Mandatory = $true, ParameterSetName = 'multi-mof')]
+        [Parameter(Mandatory = $true, ParameterSetName = 'multi-dsc')]
+        [string]
+        $checklistSTIGFiles,
 
         [Parameter()]
         [String]
-        $ManualCheckFile
+        $manualChecklistEntries,
+
+        [Parameter(Mandatory = $true)]
+        [System.IO.FileInfo]
+        $OutputPath
     )
 
     # Validate parameters before continuing
-    if ($ManualCheckFile)
+    if ($manualChecklistEntries)
+    {
+        if (-not (Test-Path -Path $manualChecklistEntries))
```

Comment:
Since $ManualCheckListEntries is a parameter we could handle this validatation in a [ValidateScript ](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-7#validatescript-validation-attribute)attribute

Reply:
The parameter isn't mandatory so I'm not sure how I would use a ValidateScript attribute.